### PR TITLE
HandleFuncs can now be Type Methods as well.

### PR DESCRIPTION
**Pull request by ivaningrooves**
`http.HandleFunc` documentation was not discovered because only
first class functions were traversed when building apib documentation.

Now both first class functions and type methods are traversed.
For example:

    type WidgetController struct {}

    // GetTypeWidget retrieves a single Widget
    func (* WidgetController) GetWidget(w, r) { }

### CODE REVIEW

**Comment by adams-sarah** [comment]
Hi! Thanks so much for the PR, I'm sorry I haven't had a chance to review your code yet, things have been busy with work and Women Who Go. I will try to review your code this weekend.

Sarah

**Comment by ivaningrooves** [comment]
Hi Sarah,

funny when you said Women Who Go at first, I thought you meant the go board
game. After googling your organization, I am very impressed with what you
are doing! I am using your library for a work project, I work with
INgrooves music in Victoria B.C. Canada. It seems my PR's tests are failing
at the moment. Let me look into this as soon as I have more time.

Cheers, Ivan

-- 
Ivan

**Comment by adams-sarah** [comment]
Yea, the tests were already broken. You can run the example pkg tests and check the output manually.

You've rewritten all the imports to `github.com/ivaningrooves/test2doc` instead of `github.com/adams-sarah/test2doc` so there's no way I can merge this ATM.

**Comment by ivaningrooves** [comment]
Yes, how would you do this if you were me? The PR contains 2 commits; the
first one is the actual change, the second involves the import rewrite so I
was able to confirm it works locally.

-- 
Ivan

**Comment by adams-sarah** [comment]
Ah, so pull requests are based off branches. So you need to get rid of the second commit in this branch.

What I would do is:
```
$ cd $GOPATH/github.com/ivaningrooves/test2doc
$ git checkout master # you're probably already on the master branch
$ git checkout -b tmp # save the current state of your master branch on a new branch, `tmp`
$ git checkout master # checkout master branch again
$ git reset --hard HEAD^ # reset backwards 1 commit from HEAD
$ git push -f origin master # force-push the new state of your master branch
```

Now if you ever want to look at the second commit again, it will be on the `tmp` branch locally.

Hope that helps!

**Comment by ivaningrooves** [comment]
Thank you!

You should be able to build & run the PR now.

-- 
Ivan

**Comment by adams-sarah** [comment]
Hm, it looks like the imports are still pointing at your code and not mine.

Did you follow the steps I outlined above? Or did you do something else

File: example/README.md [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-    └── widgets.apib
-```
-
-### 2. combine the doc files
-
-```bash
-./scripts/combine.sh
-```
-
-final apib doc file is generated at `example/apidoc.apib`
```
Comment (adams-sarah) [inline example/README.md]: why has this file been deleted?

**Comment by adams-sarah** [comment]
It looks like you deleted my example pkg too :(

**Comment by ivaningrooves** [comment]
Hi Sarah,

I did remove the changes from the PR but I had to add them back to my fork
to not break my companie's CI pipeline. I don't understand why my second
revision gets added to the PR automatically. I might have to create a new
PR... Give me moment to fix it.

-- 
Ivan

**Comment by ivaningrooves** [comment]
I created a new PR https://github.com/adams-sarah/test2doc/pull/3

**Comment by ivaningrooves** [comment]
See #3

**Pull request closed without merging**